src/postgres/getCatalog.js

```diff
@@ -75,7 +75,22 @@ const addTables = (client, catalog) =>
     select
       n.nspname as "schemaName",
       c.relname as "name",
-      d.description as "description"
+      d.description as "description",
+      case when
+           (pg_catalog.pg_relation_is_updatable(c.oid, true)::bit(8) & B'00010000') = B'00010000'
```

Comment:
We could do inserts and updates with rollbacks, but we would need to keep in mind any static fields in views, cause you will get an error about that field but the actual view is updatable as long as you dont include that field. So would there be a lot of complexity in adding that functionality?

Another option is to outer join to `information_schema.views` but there is no distinction between update and delete there: https://www.postgresql.org/docs/9.1/static/infoschema-views.html. And you would the logic to check for instead of triggers as well.

the `pg_relation_is_updatable` function was a bit hard to find , there doesnt seem to be any official documentation that is jumping out of Google at me. I found it via this: https://www.postgresql.org/message-id/CAEZATCV2_qN9P3zbvADwME_TkYf2gR_X2cLQR4R+pqkwxGxqJg@mail.gmail.com

Then I looked at the code here: https://github.com/postgres/postgres/blob/2410a2543e77983dab1f63f48b2adcd23dba994e/src/backend/utils/adt/misc.c#L684 and https://github.com/postgres/postgres/blob/3aff33aa687e47d52f453892498b30ac98a296af/src/backend/rewrite/rewriteHandler.c#L2351

It doesnt really say whether it is a private or public api, so I just assume it is public because we can access it? In any event it does seem to be the most comprehensive way to check and takes into account instead of triggers as well as auto updatable requirements. Given this, it would be my preference to keep using that - however if there is a less obscure function you would prefer let me know.